fighters/koopa/src/status.rs

```diff
@@ -31,4 +54,195 @@ pub unsafe fn exec_special_hi_a(fighter: &mut L2CFighterCommon) -> L2CValue {
     }
     WorkModule::inc_int(fighter.module_accessor, *FIGHTER_KOOPA_STATUS_SPECIAL_HI_WORK_INT_F);
     return 0.into()
+}
+
+// NEUTRAL SPECIAL
+
+#[status_script(agent = "koopa", status = FIGHTER_STATUS_KIND_SPECIAL_N, condition = LUA_SCRIPT_STATUS_FUNC_STATUS_MAIN)]
+unsafe fn special_n_main(fighter: &mut L2CFighterCommon) -> L2CValue {
+    let can_fireball = VarModule::get_int(fighter.battle_object, vars::koopa::instance::FIREBALL_COOLDOWN_FRAME) <= 0;
+    if (!can_fireball){
+        return original!(fighter);
+    }
+    else{
+        fighter.sub_change_motion_by_situation(Hash40::new("special_n_max").into(), Hash40::new("special_air_n_max").into(), false.into());
+        fighter.sub_set_ground_correct_by_situation(true.into());
+
+        fighter.sub_shift_status_main(L2CValue::Ptr(specialnmax_main_loop as *const () as _))
+    }
+}
+
+unsafe extern "C" fn specialnmax_main_loop(fighter: &mut L2CFighterCommon) -> L2CValue {
+    fighter.sub_change_motion_by_situation(Hash40::new("special_n_max").into(), Hash40::new("special_air_n_max").into(), true.into());
```

Comment:
This should go behind an `StatusModule::is_situation_changed` check, so it's not running every frame